checks/evaluation/branch_protection.go

```diff
@@ -49,6 +49,14 @@ type levelScore struct {
 	maxes  scoresInfo // Maximum possible score for a branch.
 }
 
+// Evaluates if Scorecard is being run with an administrator token.
+func isUserAdmin(branchProtectionData *clients.BranchRef) bool {
+	// When Scorecard is run without the admin token, Github retrieves both of the following fields as nil,
+	// so we're using them to evaluate if Scorecard is run using admin token or not.
+	return branchProtectionData.BranchProtectionRule.CheckRules.UpToDateBeforeMerge != nil ||
+		branchProtectionData.BranchProtectionRule.RequireLastPushApproval != nil
```

Comment:
Does this require any of these settings to be set? Or are they non-nil regardless?

Reply:
If Scorecard was run with admin permissions or with the repository using Repo rules, they would be non-nil regardless. But independent of this, I've changed this function to use information coming directly from our GitHub client, which have a more solid way to evaluate if Scorecard is run with limited permissions.